tests/mir-opt/pre-codegen/mem_swap.swap_generic.PreCodegen.after.mir

```diff
@@ -0,0 +1,132 @@
+                                let mut _12: *mut T; // in scope 12 at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                                scope 13 {
+                                    scope 14 (inlined ptr::read_mut::runtime::<T>) { // at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                                        debug src => _12; // in scope 14 at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                                    }
+                                }
+                            }
+                        }
+                        scope 9 (inlined ptr::read_mut::<T>) { // at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+                            debug src => _7; // in scope 9 at $SRC_DIR/core/src/ptr/mod.rs:LL:COL
+                            let mut _11: *mut T; // in scope 9 at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                            scope 10 {
+                                scope 11 (inlined ptr::read_mut::runtime::<T>) { // at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                                    debug src => _11; // in scope 11 at $SRC_DIR/core/src/intrinsics.rs:LL:COL
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bb0: {
+        StorageLive(_4);                 // scope 1 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        _4 = const _;                    // scope 1 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        switchInt(move _4) -> [0: bb4, otherwise: bb2]; // scope 1 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+    }
+
+    bb1: {
+        return;                          // scope 0 at $DIR/mem_swap.rs:+2:2: +2:2
+    }
+
+    bb2: {
+        StorageLive(_5);                 // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        _5 = &raw mut (*_1);             // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        StorageLive(_6);                 // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        _6 = &raw mut (*_2);             // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        _3 = swap_nonoverlapping::<T>(move _5, move _6, const 1_usize) -> bb3; // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+                                         // mir::Constant
+                                         // + span: $SRC_DIR/core/src/mem/mod.rs:LL:COL
+                                         // + literal: Const { ty: unsafe fn(*mut T, *mut T, usize) {swap_nonoverlapping::<T>}, val: Value(<ZST>) }
+    }
+
+    bb3: {
+        StorageDead(_6);                 // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        StorageDead(_5);                 // scope 2 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        StorageDead(_4);                 // scope 1 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+        goto -> bb1;                     // scope 1 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
```

Comment:
Not related to this PR; I'm just curious: why do we `goto -> bb1` instead of simply `return`?

Reply:
LLVM prefers having a single return terminator. So have a pass to replace such "go to return", but disabled by default. @saethlin did a perf run recently IIRC.

Reply:
https://github.com/rust-lang/rust/pull/106362 is the "just enable the pass" perf runs
https://github.com/rust-lang/rust/pull/106550 is enabling the pass then cleaning up after it later in codegen

I don't think this is particularly worth pursuing based on the data available in those PRs, but I have also noticed that there are cases where we can end up with multiple return terminators heading into codegen. Whether or no